### PR TITLE
Edit page settings dialog - edit slug and title

### PR DESCRIPTION
**Pull request by sy238-unikent**

### CODE REVIEW

File: resources/assets/js/components/PageListItem.vue [hunk 1]
```diff
@@ -23,7 +23,7 @@
 		</span>
 
 		<span ref="name" class="page-list__text" @click="edit">
-			{{ page.path === '/' ? 'Home page' : (page.revision.title || page.slug) }}
+			{{ page.path === '/' ? 'Home page' : (page.title || page.slug) }}
```
Comment (sy238-unikent) [inline resources/assets/js/components/PageListItem.vue]: The title is available as part of the Page data object itself. Shouldn't need to access it via the revision.

File: resources/assets/js/components/TopBar.vue [hunk 2]
```diff
@@ -167,8 +167,8 @@ Note that the page editing toolbar is a separate component found in `components/
 			backToSites() {
 				// another prompt to save the page when going back to the site listing
 				this.promptToSave(() => {
-					this.$store.commit('setPage', {});
 					this.$store.commit('setLoaded', false);
+					this.$store.commit('setPage', {});
```
Comment (sy238-unikent) [inline resources/assets/js/components/TopBar.vue]: Changing the order of this seemed to fix an intermittent issue I was having where the back to sites button stopped working as somewhere the code was trying to access page data (I think in updating the site name and url in the title bar).

File: resources/assets/js/store/index.js [hunk 3]
```diff
@@ -57,6 +57,38 @@ let store = new Vuex.Store({
 
 	mutations: {
```
Comment (sy238-unikent) [inline resources/assets/js/store/index.js]: These mutations update the data model held in the state.

They update the data both in the currently edited page (state.page.pageData) IF that is the page whose data is being updated AND the data held in state.page.pages (the site structure).

File: resources/assets/js/store/index.js [hunk 4]
```diff
@@ -57,6 +57,38 @@ let store = new Vuex.Store({
...
 
+		/**
```
Comment (sy238-unikent) [inline resources/assets/js/store/index.js]: These mutations update the data model held in the state.

They update the data both in the currently edited page (state.page.pageData) IF that is the page whose data is being updated AND the data held in state.page.pages (the site structure).

File: webpack.config.babel.js [hunk 5]
```diff
@@ -121,17 +121,6 @@ export default {
 					name: '[name].[ext]?[hash]'
 				}
 			},
-            {
```
Comment (sy238-unikent) [inline webpack.config.babel.js]: Removed this is it was a failed attempt to automatically generate documentation from the js.

**Comment by cfc7-unikent** [comment]
Looks like it is working as expected. 

Suggestion for future work - replace the console debugging error when a user tries to create a page with a slug which matches an existing slug with an error in the model telling them what to change. 

But it's a big improvement the current state so happy to merge this in as if you think so.

**Comment by sy238-unikent** [comment]
Thanks for changes @cfc7-unikent , looks good.

Please merge!

**Pull request merged**